sotodlib/preprocess/pcore.py

```diff
@@ -437,6 +443,10 @@ def run(self, aman, proc_aman=None, select=True, sim=False):
                 proc_aman.restrict('dets', det_list)
             full = proc_aman.copy()
             run_calc = False
+            if update_plot:
```

Comment:
I'm not sure `update_plot` is needed I think `run_plot` can be taken as `True` or `False` either in the `if proc_aman is None` or in the `else` block.